devtools/x/src/build.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    cargo::{CargoArgs, CargoCommand},
+    context::XContext,
+    utils, Result,
+};
+use std::ffi::OsString;
+use structopt::StructOpt;
+
+arg_enum! {
+    #[derive(Debug)]
+    enum Coloring {
+        Auto,
+        Always,
+        Never,
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    #[structopt(long, short)]
+    /// No output printed to stdout
+    quiet: bool,
+    #[structopt(long, short, number_of_values = 1)]
+    /// Package to build (see `cargo help pkgid`)
+    package: Vec<String>,
+    #[structopt(long)]
+    /// Alias for --workspace (deprecated)
+    all: bool,
+    #[structopt(long)]
+    /// Build all packages in the workspace
+    workspace: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Package to build (see `cargo help pkgid`)
+    exclude: Vec<String>,
+    #[structopt(long, short)]
+    /// Number of parallel jobs, defaults to # of CPUs
+    jobs: Option<u16>,
+    #[structopt(long)]
+    /// Build only this package's library
+    lib: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified binary
+    bin: Vec<String>,
+    #[structopt(long)]
+    /// Build all binaries
+    bins: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified example
+    example: Vec<String>,
+    #[structopt(long)]
+    /// Build all examples
+    examples: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified test target
+    test: Vec<String>,
+    #[structopt(long)]
+    /// Build all tests
+    tests: bool,
+    #[structopt(long, number_of_values = 1)]
+    /// Build only the specified bench target
+    bench: Vec<String>,
+    #[structopt(long)]
+    ///  Build all benches
+    benchs: bool,
+    #[structopt(long = "all-targets")]
```

Comment:
You probably can't drop the "all-targets" part and just leave it saying `#[structopt(long)]` this is because the default is kebab-case